src/plugins/ucx/ucx_backend.cpp

```diff
@@ -879,27 +893,43 @@ nixl_status_t nixlUcxEngine::postXfer (const nixl_xfer_op_t &operation,
         }
     }
 
+    /*
+     * Flush keeps intHandle non-empty until the operation is actually
+     * completed, which can happen after local requests completion.
+     */
     rmd = (nixlUcxPublicMetadata*) remote[0].metadataP;
     ret = uw->flushEp(rmd->conn.ep, req);
     if (_retHelper(ret, intHandle, req)) {
         return ret;
     }
 
-    if(opt_args && opt_args->hasNotif) {
-        ret = notifSendPriv(remote_agent, opt_args->notifMsg, req);
-        if (_retHelper(ret, intHandle, req)) {
-            return ret;
+    ret = intHandle->status();
+    if (opt_args && opt_args->hasNotif) {
+        if (ret == NIXL_SUCCESS) {
+            ret = notifSendPriv(remote_agent, opt_args->notifMsg, req);
+            _retHelper(ret, intHandle, req);
+        } else if (ret == NIXL_IN_PROG) {
+            intHandle->notif_schedule(remote_agent, opt_args->notifMsg);
```

Comment:
I don't think it is correct to store this information in nixlUcxBackendH.
`nixlUcxEngine::postXfer` has two independent parameters: `const std::string &remote_agent` and `nixlBackendReqH* &handle`. There is no limitation to call the method with two different agents, with the same handle. The first agent may not receive the notification.

Reply:
i do not think it is supported. Also it is existing logic

Reply:
This is an internal API, so I'd expect that NIXL core should handle that.
De-facto you are limited to use only one target on a handler at the time.

Reply:
> i do not think it is supported. Also it is existing logic

The issue with the new logic. However, if it's not supported, then this is not an issue.

Reply:
>This is an internal API, so I'd expect that NIXL core should handle that.
De-facto you are limited to use only one target on a handler at the time.

I see. Then it seems to be ok to store `remote_agent` in `intHandle`.